### ISSUE TITLE
Inconsistent Training Progress regarding epoch number

### ISSUE THREAD

**Issue by coliinkc** [ThilinaRajapakse/simpletransformers#697]
Hi,

so I used the simple transformers classification model for finetuning bert-base-multilingual on own data.
I trained the model two times using the exact same hyperparameters, only a different number of epochs (2 and 4) and evaluated every 50 steps.

For the model, that is trained for 4 epochs, the best model (saved in the best_model folder) is reached after 2 epochs already, continuing training leads to overfitting which can be seen in training_progress_scores.csv.

However, when I set the number of epochs in the hyperparameters to 2, I obtain different eval results on the dev set. 

Shouldn't the scores on the same dev set be equal after 2 epochs, no matter if the training is continued or not?

Thanks in advance!

**Comment by ThilinaRajapakse** [comment]
No, it's not going to be identical. The learning rate decay is affected by the number of total training steps, for example. The classification layers are also randomly initialized.

**Comment by coliinkc** [comment]
Thanks for the fast response and explanation. The learning rate decay being affected by the number of total training steps makes total sense, thanks! But can I ask you again about the random initialization of the classification layers you mentioned? As far as I can see, the classification layer is added in bert_model.py and I don't see a random component in there. And the random seed used in classification_model.py is determined through the given manual seed through the hyperparameters, right?

**Comment by ThilinaRajapakse** [comment]
The classification layer (or any other task-specific layer) is not a part of the pre-trained Transformer model. When you initialize a model from pre-trained weights (e.g. `roberta-base`) for a classification task, the pre-trained weights are only available for the Transformer architecture itself and not the classification layer on top of it. This means that the classification layer will be randomly initialized unless you are loading a fine-tuned model (i.e. a model where the task-specific layer has also been trained).

Yes, the `manual_seed` controls the seed for all random seeds.
